File: server/src/main/java/com/ragagent/model/Product.java
package com.ragagent.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Product {
    private String productId;
    private String title;
    private String brand;
    private String category;
    private String subCategory;
    private BigDecimal basePrice;
    private String imagePath;
    private String imageBase64;  // 商品图片 Base64 编码（data:image/jpeg;base64,...）
    private String marketingDescription;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: client/app/src/main/java/com/ragagent/model/Product.kt
package com.ragagent.model

import com.google.gson.annotations.SerializedName

data class Product(
    @SerializedName("productId") val productId: String,
    @SerializedName("title") val title: String,
    @SerializedName("brand") val brand: String,
    @SerializedName("category") val category: String,
    @SerializedName("subCategory") val subCategory: String?,
    @SerializedName("basePrice") val basePrice: Double,
    @SerializedName("imagePath") val imagePath: String?,
    @SerializedName("marketingDescription") val marketingDescription: String?
)
